### PR TITLE
[dbnode] Client borrow connection API

### PR DESCRIPTION
**Pull request by robskillington**
**What this PR does / why we need it**:

Adds API for client to borrow connection.

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note
NONE
```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note
NONE
```

### CODE REVIEW

File: src/dbnode/client/session.go [hunk 1]
```diff
@@ -629,21 +630,80 @@ func (s *session) Open() error {
...
+		// Return whether user error occurred to break or not.
+		if userErr != nil {
+			multiErr = multiErr.Add(userErr)
+		}
+	})
+	if err != nil {
+		// Route error.
+		return result, err
+	}
+	return result, nil
```
Comment (linasm) [inline src/dbnode/client/session.go:689]: `result` does not get assigned anywhere, this is probably incorrect.

File: src/dbnode/client/session.go [hunk 2]
```diff
@@ -790,7 +853,7 @@ func (s *session) clusterAvailability(
 ) (bool, error) {
 	s.state.RLock()
 	queues := s.state.queues
-	topoMap, err := s.topologyMapWithStateRLock()
+	topoMap, err := s.topologyMapWithStateRLock()``
```
Comment (linasm) [inline src/dbnode/client/session.go:856]: That's some typo, will probably not compile. Other than that, LGTM.

**Review by linasm — APPROVED** [review]

**Review by robskillington — COMMENTED** [review]
LGTM

**Review by linasm — COMMENTED** [review]
LGTM

**Pull request merged**